site/content/api/Credentials-API.md

```diff
@@ -0,0 +1,323 @@
++++
+title = "Credentials API"
+weight = 435
++++
+
+The *Credentials API* is used by *Protocol Adapters* to retrieve credentials used to authenticate *Devices* connecting to the adapter. In particular, the API supports the storage, look up and deletion of *shared secrets* which are often used by IoT devices by means of *username/password* based authentication schemes.
+<!--more-->
+
+Credentials are of a certain *type* which indicates which authentication mechanism the credentials can be used with. Each set of credentials also contains an *authentication identity* which is the identity claimed by the device during authentication. This authentication identity is usually different from the *logical* ID the device has been registered with using Hono's [Registration API]({{< ref "api/Device-Registration-API.md" >}}). Multiple sets of credentials (including arbitrary *authentication identities*) can be registered for each *logical* device ID.
```

Comment:
I was asking myself if we should have (at least optional) something like "valid-until" and "valid-from" for credentials to let the protocol adapters detect outdated (but still existing) authentication identities.
WDYT? Have you thought about that already?